sdk/include/cheri.hh

```diff
@@ -1055,27 +1119,32 @@ namespace CHERI
 		 * all other cases.
 		 */
 		template<typename U>
-		bool is_subset_of(Capability<U> other)
+		bool is_subset_of(Capability<U, IsSealed> other)
 		{
 			return __builtin_cheri_subset_test(other.ptr, ptr);
 		}
 
 		/**
 		 * Seal this capability with the given key.
 		 */
-		Capability<T> &seal(void *key)
+		[[nodiscard]] Capability<T, true> seal(void *key)
 		{
-			ptr = static_cast<T *>(__builtin_cheri_seal(ptr, key));
-			return *this;
+			return {reinterpret_cast<CHERI_SEALED(T *)>(
+			  __builtin_cheri_seal(ptr, key))};
 		}
 
 		/**
 		 * Unseal this capability with the given key.
 		 */
-		Capability<T> &unseal(void *key)
+		[[nodiscard]] Capability<T, false> unseal(void *key)
 		{
-			ptr = static_cast<T *>(__builtin_cheri_unseal(ptr, key));
-			return *this;
+#if __has_extension(cheri_sealed_pointers) &&                                  \
+  defined(CHERIOT_NO_SEALED_POINTERS)
```

Comment:
Do you mean `!defined` here as everywhere else?  I don't see `CHERIOT_NO_SEALED_POINTERS` documented anywhere, so I'm not quite sure what it's supposed to be doing...

Reply:
`CHERIOT_NO_SEALED_POINTERS` is for opting out of the typed sealed pointer things so that existing code that uses `SObj` should work.  This one is the other way around because it's specifically for using a compiler that *does* support sealing in the type system, which changes the type signature of the seal and unseal builtins.